README.md

```diff
@@ -61,4 +62,13 @@ etcdadm init --snapshot /path/to/etcd.snapshot
 ## Caveats and Limitations
 
 1. Must run as root. (This is because etcdadm creates a systemd service)
-2. Does not support etcd v2.
\ No newline at end of file
+2. Does not support etcd v2.
+3. Currently tested on Container Linux, with plans for other platforms.
+
+## Design
+
+Etcdadm is meant to simplify some of the more mundane steps of operating an etcd cluster. It will download a specific release, install the binary, configure a systemd service, and add/remove a cluster member. It is used today to operate etcd clusters in an environment with no IaaS APIs and no egress to the internet.
+
+It does not automate cluster operation. For example, if a member permanently fails, and the operator cannot invoke `etcdadm reset` on that machine, the operator must use the etcd API to delete the failed member from the list of members.
+
+On the other hand, etcdadm could be used with a higher-level controller that reconciles cluster membership against some external source of truth (e.g. an IaaS API like AWS EC2). The controller would delegate all of the tasks local to the host (installation, configuration) to etcdadm.
```

Comment:
more of question about sentence form, should there be `on one hand` before this paragraph?

Reply:
Good point, thanks. Revised.